src/core.rs

```diff
@@ -604,7 +606,14 @@ impl<T> Webauthn<T> {
         // Token Binding ID for the connection.
 
         // Verify that the rpIdHash in authData is the SHA-256 hash of the RP ID expected by the Relying Party.
-        if data.authenticator_data.rp_id_hash != self.rp_id_hash {
+        // Note that if we have an appid stored in the state and the client indicates it has used the appid extension,
+        // we check the hash against this appid instead of the Relying Party
+        let has_appid_enabled = rsp.extensions.as_ref().map(|e| e.appid);
+        if let (Some(true), Some(appid)) = (has_appid_enabled, appid) {
+            if data.authenticator_data.rp_id_hash != compute_sha256(appid.as_bytes()) {
```

Comment:
https://www.w3.org/TR/webauthn-2/#sctn-appid-extension
10.2: "When verifying the assertion, expect that the rpIdHash MAY be the hash of the AppID instead of the RP ID."

The key word here is MAY. If we have a mixed u2f + webauth credential, and we send appId then we need to expect *either* RP ID, rather than one or the other. So I think this logic may not actually be correct on reflection. 

I think it should be:


```
let appid_hash = if rsp.extensions.as_ref().map(|e| e.appid).unwrap_or(false) {
    appid.map(|id| compute_sha256(id.as_bytes()))
} else {
    None
};

// Appid Hash is now Some(Vec<u8>) if requested, None if not.
debug_assert!(rsp.extensions.as_ref().map(|e| e.appid).unwrap_or(false) == appid_hash.is_some());

if ! (
    data.authenticator_data.rp_id_hash == self.rp_id_hash ||
    Some(data.authenticator_data.rp_id_hash) == appid_hash
) {
    return Err(WebauthnError::InvalidRPIDHash);
};
```